src/Helpers/PKCE.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+namespace Auth0\SDK\Helpers;
+
+/**
+ * Class PKCE
+ *
+ * @package Auth0\SDK\Helpers
+ */
+class PKCE
+{
+    /**
+     * Generate a random string of between 43 and 128 characters containing
+     * letters, numbers and "-", ".", "_", "~", as defined in the RFC 7636
+     * specification.
+     *
+     * @param int $length Code verifier length
+     *
+     * @return string
+     *
+     * @see https://tools.ietf.org/html/rfc7636
+     */
+    public static function generateCodeVerifier(int $length = 43): string
+    {
+        if ($length < 43 || $length > 128) {
+            throw new \InvalidArgumentException(
+                'Code verifier must be crated with a minimum length of 43 characters and a maximum length of 128 characters.'
+            );
+        }
+
+        $string = '';
+
+        while (($len = strlen($string)) < $length) {
+            $size = $length - $len;
+
+            try {
+                $bytes = random_bytes($size);
+            } catch (\Exception $e) {
+                $bytes = openssl_random_pseudo_bytes($size);
```

Comment:
I would suggest to add `ext-openssl: *` as a composer requirement just to make sure the openssl extension is enabled in the PHP installation.